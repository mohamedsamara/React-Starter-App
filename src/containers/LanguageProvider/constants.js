/*
 *
 * LanguageProvider constants
 *
 */

export const SET_LOCALE = 'src/LanguageProvider/SET_LOCALE';
