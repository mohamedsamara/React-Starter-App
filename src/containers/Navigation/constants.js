/*
 *
 * Navigation constants
 *
 */

export const TOGGLE_DRAWER = 'src/Navigation/TOGGLE_DRAWER';
