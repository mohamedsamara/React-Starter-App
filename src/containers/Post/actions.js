/*
 *
 * Post actions
 *
 */

import {
  FETCH_POSTS_REQUEST,
  FETCH_POSTS_SUCCESS,
  HANLDE_POSTS_FAILURE,
  DELETE_POST_SUCCESS,
} from './constants';

export function fetchPostsRequest(isFetching) {
  return {
    type: FETCH_POSTS_REQUEST,
    isFetching,
  };
}

export function fetchPostsSuccess(posts) {
  return {
    type: FETCH_POSTS_SUCCESS,
    posts,
  };
}

export function handlePostFailure(error) {
  return {
    type: HANLDE_POSTS_FAILURE,
    error,
  };
}

export function deletePostSuccess(index) {
  return {
    type: DELETE_POST_SUCCESS,
    index,
  };
}
