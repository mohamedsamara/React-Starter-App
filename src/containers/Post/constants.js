/*
 *
 * Post constants
 *
 */

export const FETCH_POSTS_REQUEST = 'src/Post/FETCH_POSTS_REQUEST';
export const FETCH_POSTS_SUCCESS = 'src/Post/FETCH_POSTS_SUCCESS';
export const HANLDE_POSTS_FAILURE = 'src/Post/HANLDE_POSTS_FAILURE';
export const DELETE_POST_SUCCESS = 'src/Post/DELETE_POST_SUCCESS';
