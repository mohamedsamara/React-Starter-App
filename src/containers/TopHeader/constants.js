/*
 *
 * TopHeader constants
 *
 */

export const OPEN_MENU = 'src/TopHeader/OPEN_MENU';
export const CLOSE_MENU = 'src/TopHeader/CLOSE_MENU';
