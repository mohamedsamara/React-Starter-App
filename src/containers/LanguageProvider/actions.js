/*
 *
 * LanguageProvider actions
 *
 */

import { SET_LOCALE } from './constants';

export function handleSetLocale(language) {
  return {
    type: SET_LOCALE,
    locale: language,
  };
}
