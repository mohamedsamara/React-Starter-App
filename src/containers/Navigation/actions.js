/*
 *
 * Navigation actions
 *
 */

import { TOGGLE_DRAWER } from './constants';

export function handleToggleDrawer(isDrawerOpen) {
  return {
    type: TOGGLE_DRAWER,
    isDrawerOpen
  };
}
