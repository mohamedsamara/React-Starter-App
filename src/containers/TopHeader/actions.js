/*
 *
 * TopHeader actions
 *
 */

import { OPEN_MENU, CLOSE_MENU } from './constants';

export function handleOpenMenu(target) {
  return {
    type: OPEN_MENU,
    target,
  };
}

export function handleCloseMenu() {
  return {
    type: CLOSE_MENU,
  };
}
