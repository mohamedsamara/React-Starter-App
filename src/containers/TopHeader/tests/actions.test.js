import { handleOpenMenu, handleCloseMenu } from '../actions';
import { OPEN_MENU, CLOSE_MENU } from '../constants';

describe('TopHeader Actions', () => {
  it('has a type of OPEN_MENU', () => {
    const target =
      '<button class="MuiButtonBase-root-29 MuiIconButton-root-89 MuiIconButton-colorInherit-90" tabindex="0" type="button" aria-label="More" aria-haspopup="true" aria-owns="long-menu"><span class="MuiIconButton-label-94"><svg class="MuiSvgIcon-root-95" focusable="false" viewBox="0 0 24 24" aria-hidden="true" role="presentation"><path fill="none" d="M0 0h24v24H0z"></path><path d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"></path></svg></span><span class="MuiTouchRipple-root-142"></span></button>';

    const expected = {
      type: OPEN_MENU,
      target,
    };
    expect(handleOpenMenu(target)).toEqual(expected);
    expect(handleOpenMenu(target)).toMatchSnapshot();
  });

  it('has a type of CLOSE_MENU', () => {
    const expected = {
      type: CLOSE_MENU,
    };
    expect(handleCloseMenu()).toEqual(expected);
    expect(handleCloseMenu()).toMatchSnapshot();
  });
});
