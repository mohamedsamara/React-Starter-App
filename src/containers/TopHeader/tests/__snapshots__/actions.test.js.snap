// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopHeader Actions has a type of CLOSE_MENU 1`] = `
Object {
  "type": "src/TopHeader/CLOSE_MENU",
}
`;

exports[`TopHeader Actions has a type of OPEN_MENU 1`] = `
Object {
  "target": "<button class=\\"MuiButtonBase-root-29 MuiIconButton-root-89 MuiIconButton-colorInherit-90\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"More\\" aria-haspopup=\\"true\\" aria-owns=\\"long-menu\\"><span class=\\"MuiIconButton-label-94\\"><svg class=\\"MuiSvgIcon-root-95\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-142\\"></span></button>",
  "type": "src/TopHeader/OPEN_MENU",
}
`;
